src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs

```diff
@@ -43,7 +64,8 @@ public static string AddQueryString([NotNull] string uri, [NotNull] IDictionary<
                 sb.Append(UrlEncoder.Default.UrlEncode(parameter.Value));
                 hasQuery = true;
             }
-            return sb.ToString();
+
+            return sb.ToString() + anchorText;
```

Comment:
Should probably append `anchorText` to the `sb` and then call `.ToString()` on it to prevent the extra string being created.